packages/react-reconciler/src/ReactFiber.js

```diff
@@ -325,12 +334,21 @@ export function createWorkInProgress(
     }
   }
 
-  workInProgress.expirationTime = expirationTime;
+  // Don't touching the subtree's expiration time, which has not changed.
+  workInProgress.childExpirationTime = current.childExpirationTime;
+  if (pendingProps !== current.pendingProps) {
```

Comment:
Not all the fiber used this bailout protocol. What makes you sure that it is safe to assume that this hasn't changed for all types of fibers?